### ISSUE TITLE
idea: whiteboard, stroke size, color palette

### ISSUE THREAD

**Issue by hungrysnake** [pubpub-zz/ppInk#12]
whiteboard: 
it is interesting when someone asks me a quick question and I want to quickly answer it by drawing. I find an area on the screen as a temporary drawing board. I open a notepad or any application that can give me a plain color area so that I can open ppink to draw on it. It would be nice if there is a button or hotkey change the entire screen to a black/white board for drawing.

stroke size:
the current stroke size picker is slightly unintuitive. What if we use the mouse scroll wheel to instantly change the stroke size and showing it directly on the pen tip or cursor.

color palette:
just another quick way to change color in drawing mode.

thank you so much for making ppInk the best annotation tool I use everyday.

**Comment by pubpub-zz** [comment]
whiteboard:
I thought already about it : as we are working on a screen capture tool, a full white board may be a very specific case. What you have with my tools is the capability to draw first a white rectangle (Rectangle 3 times) and selecting the area and then you are good to go :)

stroke size:
Personnaly I do not use it too much, working more with predefined size (small for drawing and large semitransparent for highlighting).mouse wheel is an idea but I'm trying to keen the tool compatible with touchscreen/tablets. but it can be an option...

color palette:
Can you clarify.

Happy you enjoy it :)

**Comment by hungrysnake** [comment]
whiteboard: 
that's an interesting use of rectangle tool. just one minor problem is that the eraser may accidentally remove the rectangle if the pen stroke to be erased is too near to the edge

color palette: 
I actually mean color picker :) just another quick way of select color that may not used often.

**Comment by pubpub-zz** [comment]
Hi @hungrysnake ,
I'm working on the color palette. What I'm proposing is to access pen setup not only throught option dialog box but also with long click (>1s) on the pens or also right click. for the moment there is no notificiation that you've pressed the button for a long click( in progress)
Can you give me your feed backs about it.
[ppInk.zip](https://github.com/pubpub-zz/ppInk/files/5198134/ppInk.zip)

for the whiteboard. If you delete the rectangle inadvertantly, just use undo/ctrl+Z to get it back :)

Give me your feedback

**Comment by pubpub-zz** [comment]
I've carried on my development and integration, can you test this beta release
[ppInk.zip](https://github.com/pubpub-zz/ppInk/files/5204822/ppInk.zip)

Thanks

**Comment by hungrysnake** [comment]
Hi @pubpub-zz 
I like the way to long press the pen icon to showing the color palette with the embedded size selector.
the color picker is a surprise. a handy way to pickup color from the screen. 
![image](https://user-images.githubusercontent.com/42885440/92870563-92693080-f436-11ea-97cf-0a5220890f42.png)

other feedback
1. instead of showing the palette after releasing mouse click, it seems showing automatically after 1,2 seconds would achieve the same result. in my first attempt on using the long press, i got no idea when to release the mouse click.

2. i occasionally get a message box about t=xxxxxxx when closing the panel or >1s accessing the color palette (img attached)
it can be reproduced by selecting the cursor mode icon twice
![Snipaste_2020-09-11_13-49-44](https://user-images.githubusercontent.com/42885440/92870308-45855a00-f436-11ea-957c-505e9e040919.png)

3. the pen size selector in color palette would be nice to have the actual size preview. maybe something like this.
![image](https://user-images.githubusercontent.com/42885440/92871263-4d91c980-f437-11ea-9f13-470141affcdd.png)

thanks for making such great tools.

**Comment by hungrysnake** [comment]
> for the whiteboard. If you delete the rectangle inadvertantly, just use undo/ctrl+Z to get it back :)

i've thinking about using rect tool to set a temporary background vs having a dedicated button/key-combination to callout the drawing area.

Here is the scenario
1. When I want to draw something with ppink.
2. I open the ppink with default curve tool
3. I select the rect tool, click twice to select fill/solid mode, then draw a rect
4. I switch back to the curve tool, select pen tool for diff color to draw (quite often I either select the pen color directly and forget that I am still in rect tool, end up drawing rect on rect, or I select curve directly and forget that I'm using the same color as the rect background)
5. repeat

If there is a shortbut to instantly switch the screen to a black/white board, I could continue with the currently drawing mode without switching back and fore for the color and tools.

**Comment by pubpub-zz** [comment]
Hi @hungrysnake ,
Thanks for your comments/proposal. From 1st post. 
1: I'm surprised about your comment about 1.2 sec, as it is what I've implemented. Can you recheck on this version.
2: this was for debug and removed
3: Gink/Ppink has already such a feature: You have to activate the pentip cursor:
![image](https://user-images.githubusercontent.com/4083478/92997261-4390c900-f512-11ea-8e2b-2dfe8caa3810.png)

About Whiteboard, having a background independant is not very easy for me. What I propose is to extend clear button: once engaged it will ask you which color to use after deleting the obects: Transparent(as legacy), white, custom, black.
custom is a semi-transparent gray but can be customed through a long press on the clean button (not yet throught options)
It is still a rectangle covering the whole surface, so can be erased but undo will restore it in case of mistake.
Here you are my beta 
[ppInk.zip](https://github.com/pubpub-zz/ppInk/files/5212817/ppInk.zip)

**Comment by hungrysnake** [comment]
Thanks for the great work.
1. :) I actually type "1,2" meaning 1 or 2 seconds. Haha
2. thanks. I guess it is something related to the color value.
3. I missed that setting. However, the pen tip doesn't reflect the size change in real-time (while moving the slider). Still need to try a few times to get the right size. Using scroll wheel to adjust pen size with real-time pen tip size preview is preferred.
4. the implementation of using trash can button is good. I like that. For most people if they don't use white/black board much, the extra click to clear screen maybe a hurdle. Thinking if double-clicking the trash can button to clear screen in transparent mode maybe a workable option.

bug:
I guess the long pressing the trash can button is meant for setting the custom background color. The pen adjustment windows doesn't close after pressing the OK button.

**Comment by pubpub-zz** [comment]
Can you tell me how often you are using the bin icon?

**Comment by hungrysnake** [comment]
Quite often actually. If I continuously draw on the screen in teaching (likely using blackboard) or demo (likely transparent), I use bin icon to erase everything. 

Another situation I use bin icon when I find it difficult to erase drawing on the screen because there are too many tiny pen stroke which is very difficult to erase. Using bin icon to clear everything is much quicker.

**Comment by pubpub-zz** [comment]
Copy !
I will therefore slightly change my implementation:
short click : erase board and reapply the last selected background
long click:  erase board and select which board to apply. the customed color adjustment will be only accessible from options
In options, you will be able to select:
* background color at toolbar opening (transparent/white/customed/black / last selected) 
* customed background color

**Comment by hungrysnake** [comment]
yeah. I think that will work.
Changing custom background color may not be often. People usually use white/black board with proper pen color for enough contrast. having custom bg color in option menu is reasonable.

**Comment by pubpub-zz** [comment]
I have a new candidate version. Can you give me your feed backs
[ppInk.zip](https://github.com/pubpub-zz/ppInk/files/5213266/ppInk.zip)

**Comment by hungrysnake** [comment]
when clicking bin icon to clear in colored bg, there is a flash of desktop background showing. is it avoidable?

Seems like there is a disconnection between the bg setting in toolbar and the setting menu. 
for example, 
1. in setting menu, i have the bg set to white
2. I open ggink, the bg now is white (that's correct, good)
3. change the bg to black in toolbar, the bg now is black, using bin to clear showing black (good)
4. close ggink and reopen it, the bg now is white (it follows the bg in setting menu. fine)
5. click the bin to clear, now the bg is black (surprise, this follow the previous bg setting in toolbar)

it seems ggink first open follows the setting bg color but the subsequence bin clear follows the previous bg color setting in toolbar.

one side question. it is possible to autosave setting without using "save to file" button in setting menu?

**Comment by pubpub-zz** [comment]
I've been able to remove the flash.
I originally preferred to have the background at startup and background through bin. Based on your reaction. I've changed that : Now the background selected through bin is reset to background.

For the auto save. I had back experience where the default saved was wrongly modified by evaluation / temporary.  I'm not convinced that depressing save is too complex.

Finally, mouse wheel to change size has been implemented.
I've issed RC2
